torch_glow/src/PyTorchFileLoader.cpp

```diff
@@ -15,30 +15,189 @@
  */
 
 #include "PyTorchFileLoader.h"
+#include "PyTorchModelLoader.h"
 #include "glow/Support/Error.h"
 #include "glow/Support/Support.h"
+#include <ATen/core/grad_mode.h>
+#include <torch/csrc/jit/operator_options.h>
+#include <torch/csrc/jit/pass_manager.h>
 
 namespace glow {
 
-PyTorchFileLoader::PyTorchFileLoader(
-    const std::string &fileName,
-    std::shared_ptr<torch::jit::script::Module> &module, llvm::Error *errPtr) {
-  auto setup = [&]() -> llvm::Error {
-    try {
-      module = std::make_shared<torch::jit::script::Module>(
-          torch::jit::load(fileName));
-    } catch (const std::exception &x) {
-      RETURN_ERR(strFormat("Cannot load model from file: %s, , reason: %s",
-                           fileName.c_str(), x.what()));
+namespace {
+/// struct keeps pointers to Glow Function and Placeholders as a local thread
+/// variable.
+struct LocalFusionFunction {
+  glow::Function *function{nullptr};
+  std::vector<glow::Placeholder *> *inputPlaceholders{nullptr};
+  std::vector<glow::Placeholder *> *outputPlaceholders{nullptr};
+  const PyTorchLoaderSettings *settings{nullptr};
+
+  void set(glow::Function *f, std::vector<glow::Placeholder *> *in,
+           std::vector<glow::Placeholder *> *out,
+           const PyTorchLoaderSettings *s) {
+    function = f;
+    inputPlaceholders = in;
+    outputPlaceholders = out;
+    settings = s;
+  }
+
+  void reset() { set(nullptr, nullptr, nullptr, nullptr); }
+};
+
+/// Meyer's singleton for static fusion symbol.
+static at::Symbol getFusionSymbol() {
```

Comment:
why do we need another symbol?

Reply:
Yes, because we a registering different operators and correspondently different fuse passes.